libminifi/include/c2/ControllerSocketProtocol.h

```diff
@@ -61,6 +66,46 @@ class ControllerSocketProtocol {
   std::weak_ptr<ControllerSocketReporter> controller_socket_reporter_;
   std::shared_ptr<Configure> configuration_;
   std::shared_ptr<core::logging::Logger> logger_ = core::logging::LoggerFactory<ControllerSocketProtocol>::getLogger();
+  std::mutex initialization_mutex_;
+
+  // Some commands need to restart the controller socket to reinitialize the socket with new data for example new SSL data in case of a flow update
+  // These commands are handled on a separate thread, and while these commands are handled other incoming commands are dropped
+  class SocketRestartCommandProcessor {
```

Comment:
The reason the controller socket is inside the FlowController is because it can be effected by the flow update. If the controller socket uses SSL the SSL context may be read from the flow configuration file. If a the flow config changes it has to be restarted in case there is a change in the used SSLContextService.